src/artifacts.js

```diff
@@ -4,8 +4,41 @@ let assert  = require('assert');
 let Promise = require('promise');
 let API     = require('taskcluster-lib-api');
 let urllib  = require('url');
+let crypto  = require('crypto');
 let api     = require('./api');
 
+/**
+ * Encode a storage key for storing in the blob storage type backends We do
+ * this to get better partitioning in S3 which gives us better performance.
+ *
+ * Basically `${md5hash(taskId + runId + name).slice(0,7)}/${taskId}/${runId}/${name}`
```

Comment:
@jhford, given how random `taskId` already is... is there any value to this?

Compared to the value of being able to access artifact even if the azure table storage is lost...
I don't say I want to plan for that happen, it would make me VERY sad, and would be super super bad. But in terms of simplicity it seems to have some value to use keys `${taskId}/${runId}/${name}`.

And there is no win by putting md5 in-front, since `taskId` is random base64 encoded. It has more entropy than hexadecimal md5. Ie. uses more characters...

Note: the odds that splitting artifacts from a single task to multiple s3 partitions has any value is very small..